src/background/assessment-data-converter.ts

```diff
@@ -183,11 +183,20 @@ export class AssessmentDataConverter {
         return null;
     }
 
-    public generateFailureInstance(description: string): UserCapturedInstance {
-        return {
+    public generateFailureInstance(description: string, path: string, snippet: string): UserCapturedInstance {
+        const instance = {
```

Comment:
This works, but upto you if you want to use spread operator or not here.
```js
function generateObject(path, snippet, description) {
    const instance = {
        ...(description !== '') ? { description } : {},
        ...(path !== '') ? { selector: path } : {},
        ...(snippet !== '') ? { html : snippet } : {},
        id: 1,
    };
    console.log(instance);
}

generateObject('', '', ''); // { id: 1 }

generateObject('1', '2', '3'); // { description: '3', selector: '1', html: '2', id: 1 }
```

Reply:
added it in and things seem to be working! Much more concise